File: cmsAdminFrontend/src/app/common/service/articleTemplate.service.ts
import {Injectable} from '@angular/core';
import {HttpHeaders} from '@angular/common/http';
import {Observable} from 'rxjs';
import {ApiService} from '../../shared/services/api.service';
import {map} from 'rxjs/operators';
import {BaseCrudService} from '../../shared/services/baseCrud.service';

@Injectable({
  providedIn: 'root'
})
export class ArticleTemplateService implements BaseCrudService {
  private ARTICLE_TEMPLATE_PATH = '/articleTemplate';

  httpOptions = {
    headers: new HttpHeaders(
      {'Content-Type': 'application/json'}
    )
  };

  constructor(private apiService: ApiService) {

  }

  getAll(): Observable<any> {
    return this.apiService.get(this.ARTICLE_TEMPLATE_PATH).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  checkUrl( url): Observable<any> {
    return this.apiService.post(this.ARTICLE_TEMPLATE_PATH + '/checkUrl', url).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  create( object): Observable<any> {
    return this.apiService.post(this.ARTICLE_TEMPLATE_PATH , object).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
  update( object): Observable<any> {
    return this.apiService.put(this.ARTICLE_TEMPLATE_PATH + '/' + object.id , object).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }

  delete( id): Observable<any> {
    return this.apiService.post(this.ARTICLE_TEMPLATE_PATH + '/delete/' + id).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }

  findByUrl( url): Observable<any> {
    return this.apiService.get(this.ARTICLE_TEMPLATE_PATH + '/findByUrl/' + url).pipe(map(
      response => {
        if (response) {
          return response;
        } else {
          console.log(response);
          return null;
        }
      }
    ));
  }
}

File: CmsAdminService/src/main/java/com/demirturk/cms/service/ArticleTemplateService.java
package com.demirturk.cms.service;

import com.demirturk.cms.exception.CmsException;
import com.demirturk.cms.model.dto.ArticleTemplateDto;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ArticleTemplateService {
    /**
     *
     * @return
     * @throws CmsException
     */
    List<ArticleTemplateDto> getAll() throws CmsException;

    /**
     *
     * @param articleTemplateDto
     * @return
     * @throws CmsException
     */
    ArticleTemplateDto create(ArticleTemplateDto articleTemplateDto) throws CmsException;

    /**
     *
     * @param articleTemplateDto
     * @param id
     * @return
     * @throws CmsException
     */
    ArticleTemplateDto update(ArticleTemplateDto articleTemplateDto, Long id) throws CmsException;

    /**
     *
     * @param id
     * @return
     * @throws CmsException
     */
    boolean delete(Long id) throws CmsException;

    /**
     *
     * @param url
     * @return
     * @throws CmsException
     */
    boolean checkUrl(String url) throws CmsException;
}
